autorest/codegen/__init__.py

```diff
@@ -135,12 +139,37 @@ def _build_package_dependency() -> Dict[str, str]:
             "dependency_msrest": "msrest>=0.6.21",
         }
 
+    @staticmethod
+    def _build_security(yaml_data: Dict[str, Any]) -> Dict[str, Any]:
+        security_yaml = yaml_data.get("security")
+        security: Dict[str, Any] = dict()
+        if security_yaml and security_yaml.get("authenticationRequired"):
+            # the key word is "AADToken" and "AzureKey"
+            aad_token = set()
+            azure_key = ""
+            for scheme in security_yaml.get("schemes"):
+                if _AAD_TOKEN == scheme["type"]:
+                    aad_token.update(scheme["scopes"])
+                elif _AZURE_KEY == scheme["type"]:
+                    # only accept the last one
+                    azure_key = scheme["headerName"]
+            if aad_token:
+                security[_AAD_TOKEN] = list(aad_token)
+            if azure_key:
+                security[_AZURE_KEY] = azure_key
+        return security
```

Comment:
you should have the same behavior as current credential flags when multiple values are passed in, see here: https://github.com/Azure/autorest.python/blob/961cddb8dad10d9425c36ffcc4a4f7ab2c8c45da/autorest/codegen/models/credential_schema_policy.py#L67

Reply:
There is no security definition about policy. If there is aad_token, the default is `BearerTokenCredentialPolicy`; If azure_key, the default is `AzureKeyCredentialPolicy`. For azure_key, there may be multi values, but I think it is better to choose one instead of raise exception.

Reply:
since these correspond one to one, it's just different ways of inputting "this is how I want to authenticate", please see my below comment for more info https://github.com/Azure/autorest.python/pull/1161#discussion_r823920673